emukit/quadrature/loop/bayesian_monte_carlo_loop.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+from ...core.loop.loop_state import create_loop_state
+from ...core.loop import OuterLoop, FixedIntervalUpdater, ModelUpdater
+from ...core.parameter_space import ParameterSpace
+from ..methods import WarpedBayesianQuadratureModel
+from ..loop.quadrature_point_calculators import SimpleBayesianMonteCarloPointCalculator
+
+
+class SimpleBayesianMonteCarlo(OuterLoop):
+    """
+    This loop implements Simple Bayesian Monte Carlo.
+
+    Nodes are samples from the probability distribution defined by the integration measure.
+    If the integration measure is the standard Lebesgue measure, then the nodes are sampled uniformly in a
+    box defined by the model (reasonable box).
+
+    C.E. Rasmussen and Z. Ghahramani
+    `Bayesian Monte Carlo' Advances in Neural Information Processing Systems 15 (NeurIPS) 2003
+
+    Note: implemented as described in Section 2.1 of the paper.
+
+    Note that this method does not depend at all on past observations. Thus it is equivalent to sampling
+    all points and then fit the model to them. The purpose of the loop is convenience, as it can be
+    used with the same interface as the active and adaptive learning schemes that depend explicitly or implicitly
+    (through hyperparameters) on the previous evaluations.
+    """
+    def __init__(self, model: WarpedBayesianQuadratureModel, model_updater: ModelUpdater=None):
+        """
+        :param model: a warped Bayesian quadrature method, e.g., VanillaBayesianQuadrature
+        :param model_updater: Defines how and when the quadrature model is updated if new data arrives. Defaults to
+        FixedIntervalUpdater. If the dummy updater NoopModelUpdater is used which does not update the model, the
+        collected nodes are stored in the loop state. When using NoopModelUpdater, you can update the model after the
+        loop ran by calling model.set_data(loop_state.X, loop_state.Y); model.optimize().
```

Comment:
I feel like this part about NoopModelUpdater isn't very relevant to this particular loop. We can consider placing it somewhere in the docs, or maybe in the docstring of the named model updater, but in here it just left me scratching my head for some time trying to figure out what is it doing here

Reply:
Well, first I had NoopModelUpdater because it safes a tremendous amount of compute. The model isn't being updated, nor optimized. Then I changed it to FixedIntervalUpdater in the end because it is kind of awkward to manually set the data and optimize the model after the loop ran (it's kind of even difficult to locate the data). Especially when benchmarking, run_loop should return the trained model.

So I thought in case someone collects many samples and wants to safe time they can use NoopModelUpdater here. A hint is nice since I don't think I would find that option otherwise. But I agree that the way I wrote it it might not convey that message... What do you think, shall I rephrase or you think such hints are not necessary?

Reply:
Just mark this part of docsting clearly as a hint

Reply:
ok, done